lib/vrl/compiler/src/value/convert.rs

```diff
@@ -153,359 +151,40 @@ impl VrlValueConvert for Value {
-        }
-    }
-}
-
-impl From<Bytes> for Value {
-    fn from(v: Bytes) -> Self {
-        Value::Bytes(v)
-    }
-}
-
-impl From<Cow<'_, str>> for Value {
-    fn from(v: Cow<'_, str>) -> Self {
-        v.as_ref().into()
-    }
-}
-
-impl From<()> for Value {
-    fn from(_: ()) -> Self {
-        Self::Null
-    }
-}
-
-impl From<String> for Value {
-    fn from(v: String) -> Self {
-        Value::Bytes(v.into())
-    }
-}
-
-impl From<&str> for Value {
-    fn from(v: &str) -> Self {
-        Value::Bytes(Bytes::copy_from_slice(v.as_bytes()))
-    }
-}
-
-// Value::Boolean --------------------------------------------------------------
-
-impl Value {
-    pub fn is_boolean(&self) -> bool {
-        matches!(self, Value::Boolean(_))
-    }
-
-    pub fn as_boolean(&self) -> Option<bool> {
-        match self {
-            Value::Boolean(v) => Some(*v),
-            _ => None,
-        }
-    }
-}
-
-impl From<bool> for Value {
-    fn from(v: bool) -> Self {
-        Value::Boolean(v)
-    }
-}
-
-// Value::Regex ----------------------------------------------------------------
-
-impl Value {
-    pub fn is_regex(&self) -> bool {
-        matches!(self, Value::Regex(_))
-    }
-
-    pub fn as_regex(&self) -> Option<&Regex> {
-        match self {
-            Value::Regex(v) => Some(v),
-            _ => None,
-        }
-    }
-}
-
-impl From<Regex> for Value {
-    fn from(v: Regex) -> Self {
-        Value::Regex(v)
-    }
-}
-
-impl From<regex::Regex> for Value {
-    fn from(regex: regex::Regex) -> Self {
-        Value::Regex(regex.into())
-    }
-}
-
-// Value::Null -----------------------------------------------------------------
-
-impl Value {
-    pub fn is_null(&self) -> bool {
-        matches!(self, Value::Null)
-    }
-
-    pub fn as_null(&self) -> Option<()> {
-        match self {
-            Value::Null => Some(()),
-            _ => None,
-        }
-    }
-}
-
-impl<T: Into<Value>> From<Option<T>> for Value {
-    fn from(v: Option<T>) -> Self {
-        match v {
-            Some(v) => v.into(),
-            None => Value::Null,
-        }
-    }
-}
-
-// Value::Array ----------------------------------------------------------------
-
-impl Value {
-    pub fn is_array(&self) -> bool {
-        matches!(self, Value::Array(_))
-    }
-
-    pub fn as_array(&self) -> Option<&[Value]> {
-        match self {
-            Value::Array(v) => Some(v),
-            _ => None,
-        }
-    }
-
-    pub fn as_array_mut(&mut self) -> Option<&mut Vec<Value>> {
-        match self {
-            Value::Array(v) => Some(v),
-            _ => None,
-        }
-    }
-}
-
-impl<T: Into<Value>> From<Vec<T>> for Value {
-    fn from(v: Vec<T>) -> Self {
-        Value::Array(v.into_iter().map(Into::into).collect::<Vec<_>>())
-    }
-}
-
-impl FromIterator<Value> for Value {
-    fn from_iter<I: IntoIterator<Item = Value>>(iter: I) -> Self {
-        Value::Array(iter.into_iter().collect::<Vec<_>>())
-    }
-}
```

Comment:
I can't seem to find this one any more. Is it because they were already implemented, or did you choose to omit certain conversion methods, and if so, what heuristics did you use to decide which to keep and which to remove?